internal/core/metadata/rest_device.go

```diff
@@ -287,8 +285,8 @@ func restGetDevicesWithLabel(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	res := make([]models.Device, 0)
-	err = dbClient.GetDevicesWithLabel(&res, label)
+	var res []models.Device
+	res, err = dbClient.GetDevicesWithLabel(label)
```

Comment:
I don't think you need the `var` line here if you change the assignment on 289 to `:=`

Just breezing over this, I think there are other `var` declarations that are no longer required (like line 255) since we aren't passing pointers into these methods as parameters anymore.

Reply:
Updated file to remove unnecessary var statements.